src/linker/configuration.py

```diff
@@ -234,3 +220,100 @@ def _get_requests(environment: Dict[Any, Any], key: str) -> Dict[Any, Any]:
                         f"Assigning default value for {key} {default_key}: '{default_value}'"
                     )
         return requests
+
+    def _get_schema(self) -> Optional[PipelineSchema]:
+        """Validates the pipeline against supported schemas.
+
+        NOTE: this acts as the pipeline configurat file's validation method since
+        we can only find a matching schema if the file is valid.
+        """
+        errors = defaultdict(dict)
+        # Check that the pipeline specification contains a single "steps" outer key
+        if not len(self.pipeline) == 1 or not "steps" in self.pipeline:
```

Comment:
It's not possible for "steps" not to be in `self.pipeline`, since this gets called after `_determine_if_spark_is_required` and we throw an exception in that method if it's not present. Is there a reason we don't get/validate the pipeline schema first?

Reply:
Oh yeah, good catch. 

Regarding validating the schema first, it seemed to make sense to validate the yaml itself prior to bothering trying to match the schema (which is why we raise - `exit_with_validation_error` - before starting the `for schema in PIPELINE_SCHEMAS` loop.